forc-plugins/forc-doc/src/lib.rs

```diff
@@ -109,6 +109,7 @@ pub fn compile_html(
         None,
         &build_instructions.experimental.experimental,
         &build_instructions.experimental.no_experimental,
+        sway_core::DbgGeneration::Full,
```

Comment:
I understand that in the end for the documentation generation it doesn't make any difference, but still the `DbgGeneration::None` looks like intuitive first choice here. Why build with debug info on when it's not needed?

Reply:
If we use `DbgGeneration::None`, it is possible that `forc doc` will compile, but later on `forc build` will not, because one expression being debugged is of a type that does not implement `Debug`.

In this case, I think it makes more sense to pass the full generation.

Reply:
I still didn't go through the all of the implementation, but isn't the type check something we need no matter if we emit `__dbg(something)` to `ecal` or to `something` just passing the something through. That's the internal implementation of `__dbg()`, but on the signature side, it should always expect `T: Debug`.